loader/icd.h

```diff
@@ -152,51 +154,58 @@ void khrIcdContextPropertiesGetPlatform(
     cl_platform_id *outPlatform);
 
 // internal tracing macros
-#if 0
-    #include <stdio.h>
-    #define KHR_ICD_TRACE(...) \
+#define KHR_ICD_TRACE(...) \
+if (khrEnableTrace && strcmp(khrEnableTrace, "True") == 0) \
+{ \
     do \
     { \
         fprintf(stderr, "KHR ICD trace at %s:%d: ", __FILE__, __LINE__); \
         fprintf(stderr, __VA_ARGS__); \
-    } while (0)
+    } while (0); \
+}
```

Comment:
I would rather see something like this:
```C
if (khrEnableTrace) \
    do \
    {
        fprintf(stderr, "KHR ICD trace at %s:%d: ", __FILE__, __LINE__); \
        fprintf(stderr, __VA_ARGS__); \
    } while (0)
```
As this will keep the established usage of requiring a `;` after each macro invocation. I also simplified the test to reflect the usage of the boolean/int variable.

Reply:
Fixed